webapp/src/components/App/Studies/StudyTree/utils.ts

```diff
@@ -160,25 +172,40 @@ export function insertFoldersIfNotExist(
  * This function doesn't mutate the tree, it returns a new tree with the workspace inserted.
  *
  * @param workspace - key of the workspace
- * @param stydyTree - study tree to insert the workspace into
+ * @param studyTree - study tree to insert the workspace into
  * @returns study tree with the empty workspace inserted if it wasn't already there.
  */
-function insertWorkspaceIfNotExist(stydyTree: StudyTreeNode, workspace: string): StudyTreeNode {
+function insertWorkspaceIfNotExist(
+  studyTree: StudyTreeNode,
+  workspace: WorkspaceDTO,
+): StudyTreeNode {
   const emptyNode = {
-    name: workspace,
-    path: `/${workspace}`,
+    name: workspace.name,
+    path: `/${workspace.name}`,
     children: [],
     hasChildren: true,
+    alias: workspace.disk_name,
   };
-  if (stydyTree.children.some((child) => child.name === workspace)) {
-    return stydyTree;
+  if (studyTree.children.some((child) => isNodeMatchWorkspace(child, workspace))) {
+    return {
+      ...studyTree,
+      children: studyTree.children.map((child) => {
+        if (isNodeMatchWorkspace(child, workspace)) {
```

Comment:
Move the logic inside StudyTreeNone by giving a `workspace` prop instead of `alias` prop.